### ISSUE TITLE
diskio duplicate reports

### ISSUE THREAD

**Issue by ydinari** [influxdata/telegraf#4113]
## Bug report

diskio plugin reports multiple times with the same parameters in the same POST request. There are report s for a disk and its partitions, and each one appears twice, and sometimes more.
Note: the POST may or may not be split across TCP packets, the duplications occur in both cases.

### Relevant telegraf.conf:

[tags]
    site = "my_site"
    cluster = "my_cluster"
    role = "my_role"
    cluster_arch = "my_arch"
    color = "my_color"

[agent]
    interval = "10s"
    debug = false
    hostname = "my_hostname"
    round_interval = true
    flush_interval = "10s"
    flush_jitter = "0s"
    collection_jitter = "0s"
    metric_batch_size = 1000
    metric_buffer_limit = 10000
    quiet = false
    omit_hostname = false

[[outputs.influxdb]]
    urls = ["http://<influxdb-server>:8086"]
    database = "telegraf"
    precision = "s"

[[inputs.diskio]]

### System info:

Debian machine on amazon. Output of hostnamectl:

Static hostname: ...
Icon name: computer-vm
Chassis: vm
Machine ID: ...
Boot ID: ...
Virtualization: xen
Operating System: Debian GNU/Linux 8 (jessie)
Kernel: Linux 3.16.0-5-amd64
Architecture: x86-64

Telegraf version:
Telegraf v1.5.1 (git: release-1.5 0605af7c)

### Steps to reproduce:

1. In the example conf file, put in an actual influxdb server url
2. Run: telegraf -config telegraf.conf

### Expected behavior:

Each diskio report line should appear only once in a POST.

### Actual behavior:

See bug report

### Additional info:

Example output of ngrep:
```
-------------------------
T 2018/05/07 08:28:30.300948 <my-ip>:54061 -> <influxdb-ip>:8086 [AP]
POST /write?db=telegraf HTTP/1.1.
Host: <influxdb-server>:8086.
Transfer-Encoding: chunked.
Content-Type: text/plain; charset=utf-8.
Accept-Encoding: gzip.
.
6fa.
diskio,role=my_role,cluster_arch=my_arch,host=my_hostname,color=my_color,site=my_site,name=xvda,cluster=my_cluster io_time=276392i,iops_in_progress=0i,writes=598804i,read_bytes=478718976i,read_time=251992i,write_time=7790256i,reads=29264i,write_bytes=10977215488i,weighted_io_time=8042040i 1525681700000000000
diskio,cluster_arch=my_arch,host=my_hostname,color=my_color,site=my_site,cluster=my_cluster,role=my_role,name=xvda1 writes=0i,read_bytes=958464i,write_time=0i,weighted_io_time=32i,reads=129i,write_bytes=0i,read_time=32i,io_time=32i,iops_in_progress=0i 1525681700000000000
diskio,cluster=my_cluster,name=xvda2,role=my_role,cluster_arch=my_arch,host=my_hostname,color=my_color,site=my_site read_time=251844i,write_time=7789896i,weighted_io_time=8042072i,reads=29075i,read_bytes=477469696i,write_bytes=10977174528i,writes=598802i,io_time=276252i,iops_in_progress=0i 1525681700000000000
diskio,name=xvda,site=my_site,cluster=my_cluster,role=my_role,cluster_arch=my_arch,host=my_hostname,color=my_color reads=29264i,read_time=251992i,iops_in_progress=0i,writes=598806i,read_bytes=478718976i,write_bytes=10977235968i,write_time=7790256i,io_time=276392i,weighted_io_time=8042040i 1525681710000000000
diskio,name=xvda1,cluster=my_cluster,role=my_role,cluster_arch=my_arch,host=my_hostname,color=my_color,site=my_site reads=129i,writes=0i,read_bytes=958464i,write_bytes=0i,read_time=32i,write_time=0i,iops_in_progress=0i,io_time=32i,weighted_io_time=32i 1525681710000000000
diskio,color=my_color,site=my_site,cluster=my_cluster,role=my_role,cluster_arch=my_arch,name=xvda2,host=my_hostname weighted_io_time=8042072i,read_bytes=477469696i,read_time=251844i,io_time=276252i,write_time=7789896i,iops_in_progress=0i,reads=29075i,writes=598804i,write_bytes=10977195008i 1525681710000000000
.
```

**Comment by danielnelson** [comment]
Looking at the ngrep output, it looks like the payload contains two collection intervals 1525681700000000000 and 1525681710000000000.  There input and output plugins and not locked together, so this would be expected in several normal cases.

You can verify that all is well by check that points are not being sent multiple times (when there are no network errors).

**Comment by ydinari** [comment]
You're right, but in other examples I see the exact same output (up to tags and fields order).
Here's an output from an identical machine with 1.6.1 telegraf (v1.6.1 (git: release-1.6 bf0ab27f):

T 2018/05/10 12:21:30.320841 <my-ip>:30112 -> <influxdb-ip>:8086 [A]
...
diskio,cluster=edge1,cluster_arch=h2o,color=b,host=<my-host>,name=xvda,role=sbc,site=<my-site> weighted_io_time=8133300i,reads=30972i,writes=744658i,io_time=300132i,write_time=7880216i,iops_in_progress=0i,read_bytes=554408960i,write_bytes=12806124544i,read_time=253324i 1525954890000000000
...

T 2018/05/10 12:21:30.322745 <my-ip>:30112 -> <influxdb-ip>:8086 [A]
...
diskio,cluster=edge1,cluster_arch=h2o,color=b,host=<my-host>,name=xvda,role=sbc,site=<my-site> write_bytes=12806124544i,read_time=253324i,write_time=7880216i,io_time=300132i,weighted_io_time=8133300i,reads=30972i,read_bytes=554408960i,iops_in_progress=0i,writes=744658i 1525954890000000000
...

I saw this several times, and identical points are received multiple times by influxdb, that's what I saw first, and only then went and checked what is actually sent from the reporting machine.
Thanks

**Comment by danielnelson** [comment]
Can you attach the output of  `cat /proc/diskstats`?

**Comment by ydinari** [comment]
Yes:
```
 202       0 xvda 30972 5082 1082830 253324 840814 1085465 27500986 7911516 0 317184 8164580
 202       1 xvda1 129 455 1872 32 0 0 0 0 0 32 32
 202       2 xvda2 30783 4616 1080390 253176 840812 1085457 27500906 7911156 0 317044 8165040
```
